lib/user-data.js

```diff
@@ -0,0 +1,65 @@
+import { IdentityVerification } from './index';
+import htmlencode from 'htmlencode';
+
+export default class UserData {
+  constructor(settings) {
+    this.loggedOut = !settings.user_id && !settings.email;
+
+    if (!settings.app_id) {
+      throw new Error('You must provide an app_id in your Intercom settings');
+    }
+    if (!this.loggedOut && !settings.verificationSecret) {
+      throw new Error('You must provide your verification secret in your Intercom settings');
+    }
+
+    this.settings = settings;
+  }
+  json() {
+    const verificationSecret = this.getVerificationSecret();
+    const identifier = this.getIdentifier();
+    this.setUserHash(verificationSecret, identifier);
+    return this.escapedSettings(this.settings);
+  }
+  getVerificationSecret() {
+    const { verificationSecret } = this.settings;
+    delete this.settings.verificationSecret;
```

Comment:
If we're going to delete the secret, we should ensure json can be called multiple times by skipping `setUserHash` if `user_hash` is already set in `this.settings`

Reply:
Updated!